include/cudaq/Optimizer/Dialect/Quake/QuakeOps.td

```diff
@@ -1118,4 +1118,38 @@ def ZOp : OneTargetOp<"z", [Hermitian]> {
   }];
 }
 
+def UnitaryOp : QuakeOp<"unitary", [AttrSizedOperandSegments]> {
+  let summary = "General unitary operation specified by constant unitary matrix data";
+  let description = [{
+    This operation affects a general user-specified unitary operation, where 
+    the unitary can be specified at compile-time or at runtime via its matrix representation.
+
+    The matrix data must be provided as a row-major 1D array. It can be a 
```

Comment:
Should we document the endian convention here as well along with the row-major convention?

Reply:
Good point! 
@boschmitt: Thoughts on how to document the default qubit ordering?